elements/rh-site-status/test/rh-site-status.spec.ts

```diff
@@ -0,0 +1,89 @@
+import { expect, html } from '@open-wc/testing';
+import { createFixture } from '@patternfly/pfe-tools/test/create-fixture.js';
+import { RhSiteStatus } from '@rhds/elements/rh-site-status/rh-site-status.js';
+
+
+function overrideFetch(ok: boolean, status: number, statusText: string, json?: () => Promise<unknown>) {
+  window.fetch = new Proxy(window.fetch, {
+    apply: (target, thisArg, args) => {
+      if (args[0] === 'https://status.redhat.com/index.json') {
+        return Promise.resolve({
+          ok,
+          status,
+          statusText,
+          json
+        });
+      }
+      return target.apply(thisArg, args);
+    }
+  });
+}
+
+
+describe('<rh-site-status>', function() {
+  describe('simply instantiating', function() {
+    let element: RhSiteStatus;
+    it('should upgrade', async function() {
+      element = await createFixture<RhSiteStatus>(html`<rh-site-status></rh-site-status>`);
+      const klass = customElements.get('rh-site-status');
+      expect(element)
+        .to.be.an.instanceOf(klass)
+        .and
+        .to.be.an.instanceOf(RhSiteStatus);
+    });
+  });
+
+  describe('when endpoint returns an All systems operational', function() {
+    let element: RhSiteStatus;
+
+    beforeEach(async function() {
+      overrideFetch(true, 200, 'All systems operational', () => Promise.resolve({ status: { indicator: 'none', description: 'All systems operational' } }));
```

Comment:
Use Sinon for these and make sure to restore the stubs after each test

Reply:
I gave this an attempt, but I couldn't wrap my head around trying to stub a private method.  I'd like to pair on this an see the approach you might take.

Reply:
sounds good

sommething like this

```js
import { stub, type Stub } from 'sinon';

describe('site status', function() {
  let fetchStub: Stub;
  beforeEach(function() {
    fetchStub = stub(window, 'fetch');
  });
  afterEach(function() {
    fetchStub?.restore();
  });
  it('fetches', function() {
    expect(fetchStub).to.have.been.calledWith(URL, options);
  });
});
```